frontend/src/components/TableContainer.tsx

```diff
@@ -11,16 +11,50 @@ import ParameterView from "./ParameterView";
 import GuardrailView from "./GuardrailView";
 import useStore from "../store";
 
+import {
+  mapCurrentJsonStateToExportParams
+} from "../utils/mapper";
+
 export default function BasicTable() {
-  const { currentTab, changeSelectedTab } = useStore(state => ({
+  const { currentTab, currentJsonState, validationResults, changeSelectedTab } = useStore(state => ({
     currentTab: state.currentTab,
+    currentJsonState: state.currentJsonState,
+    validationResults: state.validationResults,
     changeSelectedTab: state.changeSelectedTab,
   }));
```

Comment:
If there isn't any change in the mapping, we can get the properties from the store like this:
```suggestion
  const { currentTab, currentJsonState, validationResults, changeSelectedTab } = useStore();
```

Reply:
Based on looking into the pros and cons of both options I believe using the selector function inside of this hook actually provides the best performance optimizations. When using the useStore hook without the selector function then the component will re-render anytime anything in the store changes, this could result in unnecessary re-renders. If the selector function is used then only the specified parts of the store will trigger a re-render. I think all of the useStore hooks should be refactored to include the selector function. 

Selector functions are recommended both the Zustand and the Redux docs. Redux recommends further optimizations by doing things like memoizing selectors but I did not do that.
1. Zustand docs https://docs.pmnd.rs/zustand/guides/auto-generating-selectors
2. Redux docs https://redux.js.org/usage/deriving-data-selectors
3. Useful real time demo https://www.youtube.com/watch?v=DCCCcxZkIB4